### ISSUE TITLE
Possible false positives using Objects.requireNonNull?

### ISSUE THREAD

**Issue by ZOlbrys** [uber/NullAway#276]
Hi there,

Consider the following code:

```
@Nullable
CustomValue value;    // member variable
...

public void doSomething() {
    value = dao.getValueWithId(1);    // getValue can return null if no object with the specified id exists
    Objects.requireNonNull(value, "Value is null!");    // throw NPE if value is null.  Afterwards, value can be considered non-null

    final String name = value.getName();
    ...
}
```

I see the following reports when running nullaway:

```
/Users/zach/Developer/Code/sampleapp/app/src/main/java/com/zao/sampleapp/SampleClass.java:20: warning: [NullAway] dereferenced expression value is @Nullable
        final String name = value.getName();
                                 ^
    (see http://t.uber.com/nullaway )
```

Is this a false positive? Should this be safely ignored?  Or am I using `Objects.requireNonNull` incorrectly?  I know, for instance, that `value = Objects.requireNonNull(dao.getValueWithId(1))` would avoid this, but I'd like to avoid "nesting" the calls.

Edit: I am using nullaway 0.6.4 - setup via errorprone in an Android project, FWIW.

Edit2: Updated `Objects.requireNonNull` call to include a message.  This works perfectly when the message parameter is omitted.

**Comment by msridhar** [comment]
Are you on the latest release? I’m pretty sure this shouldn’t happen there

**Comment by ZOlbrys** [comment]
@msridhar my apologies for not including the version number - I am on 0.6.4, and I've included that in the original post too.

**Comment by msridhar** [comment]
@ZOlbrys I added the following code to a regression test:
```java
  static class ReqNonNullTest {

    @Nullable Object value;

    void test() {
      Objects.requireNonNull(value);
      value.toString();
    }
  }
```
I don't get any NullAway warning and it seems to be the same as your posted example.  Can you give a more detailed example where you get the warning?  A GitHub repo that shows the problem would be ideal.

**Comment by ZOlbrys** [comment]
Here's a sample project showing the issue: https://github.com/ZOlbrys/ErrorProneExample

I made an unfortunate oversight in my original example - I think the trick here is that `Objects.requireNonNull(param)` works OK.  But `Objects.requireNonNull(param, message)` does not.  Apologies for using the wrong method in my original example.  I removed the message in my example to hide project information, and then forgot to include a dummy one.  I've updated the original post to reflect this as well.  

Thanks @msridhar !

**Comment by msridhar** [comment]
Ah, that's the key!  Yeah, we can add support for the other `requireNonNull` signature, should be easy.

**Comment by ZOlbrys** [comment]
@msridhar glad to see this closed! That was quick :) Do you have an estimate on when this would be released?

**Comment by msridhar** [comment]
I don't object to cutting a new release with this fix.  @lazaroclapp WDYT?

**Comment by lazaroclapp** [comment]
We should definitely be cutting a new release soon. I was just trying to cut one tonight, but there are actually a few things I found that need fixing before that (from running on our internal codebases). I'll target this week for a 0.6.5 release, though.

**Comment by lazaroclapp** [comment]
@ZOlbrys FYI, NullAway 0.6.5 released today. Sorry for the delay!

**Comment by ZOlbrys** [comment]
The new version works great.  Thank you so much @msridhar and @lazaroclapp! 💯
